be/src/io/fs/file_system.h

```diff
@@ -52,6 +53,9 @@ class FileSystem {
 
     virtual Status create_file(const Path& path, FileWriterPtr* writer) = 0;
 
+    virtual Status open_file(const Path& path, const CacheOptions& cache_options,
```

Comment:
ReaderOption is better, open_file will create a FileReader，CacheOptions is not suitable. You can make CacheOptions as an element of ReaderOption.

Reply:
Done